### ISSUE TITLE
URLs returned by /search signed twice with token authentication

### ISSUE THREAD

**Issue by jojoob** [opencast/opencast#3387]
**Describe the bug**
I configured "Stream Security"/token based authentication on an Opencast 11 distributed setup on Docker using the [minimal configuration example](https://docs.opencast.org/r/11.x/admin/#configuration/stream-security/#minimal-configuration-example) from the docs. Everything works when doing the manual [testing](https://docs.opencast.org/r/11.x/admin/#configuration/stream-security/#testing) steps. But engage UI does not work: thumbs and videos are not loading because the URLs delivered by the "/search/" API are signed twice. URLs delivered by the external API (`curl 'https://opencastadmin.htw-berlin.de/api/events?sign=true&withpublications=true'`) are signed correct.

![ksnip_20220127-112108](https://user-images.githubusercontent.com/4180675/151339730-a3bc507f-e946-4315-9854-40a652778867.png)

`curl https://opencast.htw-berlin.de/search/episode.json`:

```json
[...]
                            {
                            "id": "4b3132c7-ba44-4700-bf3c-aebbbe7517bb",
                            "type": "presentation/player+preview",
                            "ref": "track:track-5",
                            "mimetype": "image/jpeg",
                            "tags":
                            {
                                "tag": "engage-download"
                            },
                            "url": "https://ocdownload.htw-berlin.de/static/mh_default_org/engage-player/d1c1f0e4-574a-4105-8e54-d3239b1d8ed4/82f65169-6327-49da-995b-033debdb3899/file_example_MP4_480_1_5MG_3_000s_player.jpg?policy=eyJTdGF0ZW1lbnQiOnsiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6MTY0MzI4NjIyOTcxMX0sIlJlc291cmNlIjoiaHR0cHM6XC9cL29jZG93bmxvYWQuaHR3LWJlcmxpbi5kZVwvc3RhdGljXC9taF9kZWZhdWx0X29yZ1wvZW5nYWdlLXBsYXllclwvZDFjMWYwZTQtNTc0YS00MTA1LThlNTQtZDMyMzliMWQ4ZWQ0XC84MmY2NTE2OS02MzI3LTQ5ZGEtOTk1Yi0wMzNkZWJkYjM4OTlcL2ZpbGVfZXhhbXBsZV9NUDRfNDgwXzFfNU1HXzNfMDAwc19wbGF5ZXIuanBnIn19&keyId=default&signature=aeda95221df35f5813ecea64367c88ce08f473b3f5f69526c9101c7d58db59b2&policy=eyJTdGF0ZW1lbnQiOnsiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6MTY0MzI4NjIyOTcxOH0sIlJlc291cmNlIjoiaHR0cHM6XC9cL29jZG93bmxvYWQuaHR3LWJlcmxpbi5kZVwvc3RhdGljXC9taF9kZWZhdWx0X29yZ1wvZW5nYWdlLXBsYXllclwvZDFjMWYwZTQtNTc0YS00MTA1LThlNTQtZDMyMzliMWQ4ZWQ0XC84MmY2NTE2OS02MzI3LTQ5ZGEtOTk1Yi0wMzNkZWJkYjM4OTlcL2ZpbGVfZXhhbXBsZV9NUDRfNDgwXzFfNU1HXzNfMDAwc19wbGF5ZXIuanBnP3BvbGljeT1leUpUZEdGMFpXMWxiblFpT25zaVEyOXVaR2wwYVc5dUlqcDdJa1JoZEdWTVpYTnpWR2hoYmlJNk1UWTBNekk0TmpJeU9UY3hNWDBzSWxKbGMyOTFjbU5sSWpvaWFIUjBjSE02WEM5Y0wyOWpaRzkzYm14dllXUXVhSFIzTFdKbGNteHBiaTVrWlZ3dmMzUmhkR2xqWEM5dGFGOWtaV1poZFd4MFgyOXlaMXd2Wlc1bllXZGxMWEJzWVhsbGNsd3ZaREZqTVdZd1pUUXROVGMwWVMwME1UQTFMVGhsTlRRdFpETXlNemxpTVdRNFpXUTBYQzg0TW1ZMk5URTJPUzAyTXpJM0xUUTVaR0V0T1RrMVlpMHdNek5rWldKa1lqTTRPVGxjTDJacGJHVmZaWGhoYlhCc1pWOU5VRFJmTkRnd1h6RmZOVTFIWHpOZk1EQXdjMTl3YkdGNVpYSXVhbkJuSW4xOSZrZXlJZD1kZWZhdWx0JnNpZ25hdHVyZT1hZWRhOTUyMjFkZjM1ZjU4MTNlY2VhNjQzNjdjODhjZTA4ZjQ3M2IzZjVmNjk1MjZjOTEwMWM3ZDU4ZGI1OWIyIn19&keyId=default&signature=39a037d383941310fb18eb52565b075380efc93212cf467c05d72d3dd010d43c",
                            "size": 20627
                        },
[...]
```

**To Reproduce**
1. Disable static resource authentication by setting `authentication.required = false` in `etc/org.opencastproject.fsresources.StaticResourceServlet.cfg`
2. Configure "Stream Security" on Opencast 11 as described [here](https://docs.opencast.org/r/11.x/admin/#configuration/stream-security/#minimal-configuration-example)
3. Verify correct setup by doing the [testing steps](https://docs.opencast.org/r/11.x/admin/#configuration/stream-security/#testing)
4. Open the engage UI in browser see that thumbnails and videos fail to load
5. Open browser dev tools and check the cause of failed resources. You should see that URLs are signed twice and the URL verification fails (Status Code 400)
6. Verify that URLs are signed twice by `curl https://opencast.htw-berlin.de/search/episode.json`

**Environment Information 1**
 - OS: Ubuntu 20.04
 - Docker Version: 20.10.7-0ubuntu5~20.04.2
 - Browser Chromium
 - Opencast version 11.3 Docker images
 
**Environment Information 2**
 - OS: Ubuntu 20.04
 - Browser Chromium
 - Opencast 11.4.0 - 248a5fc - built on 2022-02-23

<details>
  <summary>etc/org.opencastproject.security.urlsigning.filter.UrlSigningFilter.cfg</summary>

```cfg
# The UrlSigningFilter is a verification component that works with the UrlSigningVerifierImpl to protect Opencast
# internal resources. The filter uses a set of regular expressions to determine which requests to an Opencast instance
# need to be verified.

# Enable or disable the UrlSigningFilter. If disabled no internally provided files will be protected by Stream Security
#enabled=false
enabled=true

# Enable or disable strict checking of resources. Strict checking includes verifying that the entire requested URL
# matches the one in the policy. Non-strict checking will only verify that the path to the resource matches the policy.
# This is useful if requests for signed material is failing due to hostname changes due to redirects, players adding
# extra URL information such as ports etc.
strict=true

# This configuration file defines the urls that will be protected by the UrlSigningFilter filter. These endpoints will
# require a signed url by stream security before a file can be downloaded

# Protects: /files/collection/{collectionId}/{fileName}
url.regex.collection=.*files\/collection\/.*

# Protects: /files/mediapackage/{mediaPackageID}/{mediaPackageElementID}
# and Protects: /files/mediapackage/{mediaPackageID}/{mediaPackageElementID}/{fileName}
url.regex.mediapackage=.*files\/mediapackage\/.*

# Protects: /staticfiles/{uuid} but not /staticfiles/{uuid}/url
url.regex.staticfiles=(?\=(.*staticfiles.*))(?=^(?!.*staticfiles.*url|.*docs.*).*$)(.*)

# Protects: /archive/archive/mediapackage/{mediaPackageID}/{mediaPackageElementID}/{version}
url.regex.archive=.*archive\/archive\/mediapackage\/.*\/.*\/.*

# Protects files hosted by the built in jetty server for engage
url.regex.engage=.*static\/engage-player.*

# Protects files hosted by the built in jetty server for previews / editing
url.regex.internal=.*static\/internal.*

# According to https://docs.opencast.org/r/11.x/admin/#configuration/stream-security/
url.regex.files=.*ocdownload.htw-berlin.de/static/.*
```
</details>

<details>
  <summary>etc/org.opencastproject.security.urlsigning.provider.impl.GenericUrlSigningProvider.cfg</summary>

```cfg
key.default.secret=6EDB5EDDCF994B7432C371D7C274F
key.default.url=https://ocdownload.htw-berlin.de/static/
```
</details>

<details>
  <summary>etc/org.opencastproject.security.urlsigning.verifier.impl.UrlSigningVerifierImpl.cfg</summary>

```cfg
key.default=6EDB5EDDCF994B7432C371D7C274F
```
</details>

**Comment by jojoob** [comment]
I can confirm that Opencast 11.4.0 - 248a5fc - built on 2022-02-23 installed on Ubuntu 20.04 via package is also affected.

**Comment by jojoob** [comment]
@mliradelc any news on this? Is there something I can help with?

**Comment by mliradelc** [comment]
Hi, sorry for still not looking this, I just got married a few weeks ago and I'm taking back all my duties.

Well, I think the problem with double signing maybe comes from my PR, in the meantime you can look at #3058, We haven't any issues with that here, I recommend you to start there.

**Comment by mliradelc** [comment]
@jojoob Last week was submitted this PR #3674 , I believe that will solve your issue. Can you review it?

**Comment by jojoob** [comment]
Hi @mliradelc I cherry-picked the patch onto the code of Opencast version 11.6: I can confirm it fixes the double signing problem. :+1: 

PS: The API also still delivers correctly signed URLs. So no new problems here.

**Comment by mliradelc** [comment]
Solved with #3674
